source/vibe/inet/webform.d

```diff
@@ -453,23 +469,51 @@ unittest {
 		int opApply(int delegate(string key, string value)) { return 0; }
 	}
 	static assert(isStringMap!M);
+	static assert(isStringMapWithOpApply!M);
+	static assert(!isStringMapWithByKeyValue!M);
+
+	static struct R {
+		struct F { string key, value; }
+		F front;
+		bool empty;
+		void popFront() {};
+	}
+	static struct M2 {
+		R byKeyValue() { return R(); }
+	}
+	static assert(isStringMap!M2);
+	static assert(isStringMapWithByKeyValue!M2);
+	static assert(!isStringMapWithOpApply!M2);
 }
 
-private template isJsonLike(T)
-{
-	enum isJsonLike = __traits(compiles, () {
-		import std.conv;
-		string r;
-		foreach (string key, value; T.init)
-			r = value.type == T.Type.string ? value.get!string : value.to!string;
-	} ());
-}
+private enum isJsonLikeWithOpApply(T) = __traits(compiles, () {
+	import std.conv;
+	foreach (string key, T value; T.init)
+		string r = value.type == T.Type.string ? value.get!string : value.to!string;
+} ());
+
+private enum isJsonLikeWithByKeyValue(T) = __traits(compiles, () {
+	import std.conv;
+	foreach (string key, T value; T.init.byKeyValue())
+		string r = value.type == T.Type.string ? value.get!string : value.to!string;
+} ());
```

Comment:
```suggestion
private enum isJsonLikeWithByKeyValue(T) = __traits(compiles, () {
	import std.conv;
	foreach (item; T.init.byKeyValue())
		string k = item.key, r = item.value.type == T.Type.string ? item.value.get!string : item.value.to!string;
} ());
```

For some reason the latest DMD hangs when using the separate key/value foreach syntax. Changing this here and also at the two places where the actual iteration happens fixed this for me.

Reply:
Oh neat! If I’d known that was valid I would’ve used that all along.
Unfortunately I’m away from my computer until tomorrow but if you want to push to my fork to expedite things then feel free. Otherwise I can do it sometime tomorrow.

Reply:
Done!